test/unit/Governor.js

```diff
@@ -0,0 +1,558 @@
+            await truffleAssert.reverts(
+                governor.transferOwnership(accounts[1]),
+                "unauthorized"
+            )
+        })
+
+        it("reverts if the new owner address is the zero-value for the address type", async () => {
+            const txData = utils.arrayify(governor.contract.methods.transferOwnership(constants.AddressZero).encodeABI())
+            await governor.stage(
+                {
+                    target: [governor.address],
+                    value: ["0"],
+                    data: [txData]
+                },
+                "0"
+            )
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [governor.address],
+                        value: ["0"],
+                        data: [txData]
+                    }
+                ),
+                "newOwner is zero"
+            )
+        })
+
+        it("updates ownership to a new owner", async () => {
+            const txData = utils.arrayify(governor.contract.methods.transferOwnership(accounts[1]).encodeABI())
+            await governor.stage(
+                {
+                    target: [governor.address],
+                    value: ["0"],
+                    data: [txData]
+                },
+                "0"
+            )
+
+            const tx = await governor.execute(
+                {
+                    target: [governor.address],
+                    value: ["0"],
+                    data: [txData]
+                }
+            )
+
+            truffleAssert.eventEmitted(tx, "OwnershipTransferred", e => e.previousOwner == accounts[0] && e.newOwner == accounts[1])
+        })
+    })
+
+    describe("stageUpdate", () => {
+        it("reverts when sender is not owner", async () => {
+            const {data} = await setUint256Tx("0", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    },
+                    "0",
+                    {from: accounts[1]}
+                ),
+                "unauthorized",
+            )
+        })
+
+        it("reverts when the current block number added by the delay overflows", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            await fixture.rpc.mine()
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    },
+                    constants.MaxUint256
+                )
+            )
+        })
+
+        it("stage emits an UpdateStaged event", async () => {
```

Comment:
Let's also check that the contract storage mapping for updates is updated correctly?

Reply:
slightly annoying due to not being able to use `ethers.utils.solidityKeccak256`  . It only supports packed data but our data type doesn't support packed data. 

The least annoying solution would be to add a pure function to the governor contract (or a seperate one we just deploy for testing) that can calculate the hash for an update given the pre-image.